src/cartesian/Bar.tsx

```diff
@@ -194,6 +228,16 @@ function BarRectangles(props: BarRectanglesProps) {
 
   const { index: activeIndex } = useTooltipContext();
 
+  const onMouseEnterFromContext = useMouseEnterItemDispatch();
```

Comment:
why do we do all of this twice? Seems redundant. Is it so we're calling the events if we're in the background or the actual bar?

Reply:
Because both bars and backgrounds react to mouse events, and this is the only place that's functional component and allows hooks and so this is here twice.

There's some opportunity to DRY, I will iterate in next PR.